synfig-studio/src/synfigapp/actions/valuedescmakeparenttoactive.cpp

```diff
@@ -0,0 +1,160 @@
+ACTION_SET_LOCAL_NAME(Action::ValueDescMakeParentToActive,N_("Make Parent To Active Bone"));
+ACTION_SET_TASK(Action::ValueDescMakeParentToActive,"make_parent_to_active");
+ACTION_SET_CATEGORY(Action::ValueDescMakeParentToActive,Action::CATEGORY_VALUEDESC);
+ACTION_SET_PRIORITY(Action::ValueDescMakeParentToActive,0);
+ACTION_SET_VERSION(Action::ValueDescMakeParentToActive,"0.0");
+ACTION_SET_CVS_ID(Action::ValueDescMakeParentToActive,"$Id$");
+
+/* === G L O B A L S ======================================================= */
+
+/* === P R O C E D U R E S ================================================= */
+
+/* === M E T H O D S ======================================================= */
+
+Action::ValueDescMakeParentToActive::ValueDescMakeParentToActive():
+	time(0)
+{
+}
+
+Action::ParamVocab
+Action::ValueDescMakeParentToActive::get_param_vocab()
+{
+	ParamVocab ret(Action::CanvasSpecific::get_param_vocab());
+
+	ret.push_back(ParamDesc("value_desc",Param::TYPE_VALUEDESC)
+		.set_local_name(_("ValueDesc on parent Bone"))
+	);
+	ret.push_back(ParamDesc("time",Param::TYPE_TIME)
+		.set_local_name(_("Time"))
+		.set_optional()
+	);
+	ret.push_back(ParamDesc("active_bone",Param::TYPE_VALUENODE)
+						  .set_local_name(_("ValueNode of Active Bone"))
+	);
+
+	return ret;
+}
+
+bool
+Action::ValueDescMakeParentToActive::is_candidate(const ParamList &x)
+{
+	ParamList::const_iterator i;
+	
+	i = x.find("value_desc");
+	if (i == x.end()) return false;
+
+	ValueDesc value_desc(i->second.get_value_desc());
+	//ValueDesc value_desc(x.find("value_desc")->second.get_value_desc());
+	i=x.find("active_bone");
+	if(i==x.end()) return false;
+
+	ValueNode::Handle active_bone(i->second.get_value_node());
+	if (!candidate_check(get_param_vocab(),x))
+		return false;
+
+	return value_desc.parent_is_value_node()
+		&& ValueNode_Bone::Handle::cast_dynamic(value_desc.get_parent_value_node())
+		&& active_bone
+		&& ValueNode_Bone::Handle::cast_dynamic(active_bone);
+}
+
+bool
+Action::ValueDescMakeParentToActive::set_param(const synfig::String& name, const Action::Param &param)
+{
+	if (name == "value_desc" && param.get_type() == Param::TYPE_VALUEDESC
+	 && param.get_value_desc().parent_is_value_node()
+	 && ValueNode_Bone::Handle::cast_dynamic(param.get_value_desc().get_parent_value_node()) )
+	{
+		value_desc = param.get_value_desc();
+		return true;
+	}
+
+	if(name=="active_bone" && param.get_type()==Param::TYPE_VALUENODE
+	&& ValueNode_Bone::Handle::cast_dynamic(param.get_value_node())){
+		active_bone = param.get_value_node();
+		return true;
+	}
+
+	if(name=="time" && param.get_type()==Param::TYPE_TIME)
+	{
+		time=param.get_time();
+		return true;
+	}
+
+	return Action::CanvasSpecific::set_param(name,param);
+}
+
+bool
+Action::ValueDescMakeParentToActive::is_ready()const
+{
+	if (!value_desc)
+		return false;
+	return Action::CanvasSpecific::is_ready();
+}
+
+void
+Action::ValueDescMakeParentToActive::prepare()
```

Comment:
Does it work?

Action::Super  is supposed to be a superset of regular actions. `prepare()` is used to build the action chain.
I don't see you doing nothing here. What am I missing?

Reply:
Actually this action is still work under progress. I'm currently working on adding support to deformation layer.

Reply:
Oh okay. But why should it support multiple actions here?
Isn't it a single action?

Reply:
Yes it is.. I'll make the necessary changes